core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricUtils.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json._
+import scala.util.matching.Regex
+import scala.io.Source
+
+case class TokenServiceConfig(tokenServiceEndpoint: String,
+                              clusterType: String,
+                              clusterName: String,
+                              sessionToken: String)
+
+object TokenServiceConfigProtocol extends DefaultJsonProtocol {
+  implicit val TokenServiceConfigFormat: RootJsonFormat[TokenServiceConfig] = jsonFormat4(TokenServiceConfig)
+}
+
+import TokenServiceConfigProtocol._
+
+object FabricUtils {
+  private var TridentContext: Map[String, String] = Map[String, String]()
+
+  def getFabricContext: Map[String, String] = {
+    if (TridentContext.nonEmpty) {
+      TridentContext
+    } else {
+      try {
+        val contextFile = scala.io.Source.fromFile(FabricConstants.ContextFilePath)
```

Comment:
What exactly is this file and how does it get there?

Reply:
These files will be provision in the cluster for us by the spark and bbc-vhd team.